recipe/2023-11-support-insecure-registries.md

```diff
@@ -0,0 +1,185 @@
+```
+I would like to see `rad bicep publish` command successfully publish bicep file to locally hosted registry and also I should be able to read it from the registry during recipe deployment. 
+
+
+## Design
+
+### Design details
+Since we use ORAS client to communicate with the registry, it provides an option "plain-http" to allow insecure connections to registry without SSL check. `plainHttp` property should be set to `true` when client to the remote repository is created to support using insecure registries.
+
+#### Introduce "insecure-http" opt-in flag for users.
+
+Users can use `insecure-http` flag with `rad bicep publish` command and `insecureHttp` property while defining bicep recipes to specify the registry used is insecure and allow connections to registry without SSL check. And based on this property `plainHttp` option is set on the remote repository client. 
+
+
+
+### API design (if applicable)
+
+***Model changes***
+
+Addition of optional property `insecureHttp` to BicepRecipeProperties (as it's only valid for template kind `bicep`).
+
+environments.tsp
+```diff
+@doc("Represents Bicep recipe properties.")
+model BicepRecipeProperties extends RecipeProperties {
+  @doc("The Bicep template kind.")
+  templateKind: "bicep";
+
++  @doc("Allows an insecure connection to a Bicep registry without doing an SSL check. Used commonly for locally-hosted registries. Defaults to false (require SSL).")
++  insecureHttp?: boolean;
+}
+
+@doc("The properties of a Recipe linked to an Environment.")
+model RecipeGetMetadataResponse {
+  @doc("The format of the template provided by the recipe. Allowed values: bicep, terraform.")
+  templateKind: string;
+
+  @doc("The path to the template provided by the recipe. Currently only link to Azure Container Registry is supported.")
+  templatePath: string;
+
+  @doc("The version of the template to deploy. For Terraform recipes using a module registry this is required, but must be omitted for other module sources.")
+  templateVersion?: string;
+
+  @doc("The key/value parameters to pass to the recipe template at deployment.")
+  parameters: {};
+
++  @doc("Allows an insecure connection to a Bicep registry without doing an SSL check. Used commonly for locally-hosted registries. Defaults to false (require SSL).")
++  insecureHttp?: boolean;
+}
+```
+
+***CLI changes***
+
+Add insecure-http flag to allow insecure connection to a Bicep registry without doing an SSL check.
+#### Bicep Publish
+
+```
+rad bicep publish --file ./redis-test.bicep --target br:localhost:5000/myregistry/redis-test:v1 --insecure-http
+```
+#### Recipe Register 
+
+```
+rad recipe register cosmosdb --template-kind bicep --template-path br:localhost:5000/myregistry/redis-test:v1 --resource-type Applications.Datastores/mongoDatabases --insecure-http
+```
+
+#### Recipe List/Show
+Add a new column to recipe list/show output table.
+
+| NAME        | TYPE           | TEMPLATE KIND  | TEMPLATE VERSION | TEMPLATE | INSECURE-HTTP|
```

Comment:
I'm worried that we might be scaling too the width of the table too far.

I think we can remove it from `rad recipe list` for the table format.

And for `rad recipe show` we may only want to show it on `rad recipe show -o json`

Reply:
👍 

I'm not sure there's a need to show this in `rad recipe list/show`. There's a limit to what what we can do with tables. If users need to see every detail, that's what `-o json` is for.